src/mode.cpp

```diff
@@ -837,27 +837,35 @@ std::string ModeParser::GiveModeList(ModeMasks m)
 	return type1 + "," + type2 + "," + type3 + "," + type4;
 }
 
+struct PrefixModeSorter
+{
+	bool operator()(ModeHandler* lhs, ModeHandler* rhs)
+	{
+		return lhs->GetPrefixRank() < rhs->GetPrefixRank();
+	}
+};
+
 std::string ModeParser::BuildPrefixes(bool lettersAndModes)
 {
 	std::string mletters;
 	std::string mprefixes;
-	std::map<int,std::pair<char,char> > prefixes;
+	std::vector<ModeHandler*> prefixes;
 
 	for (unsigned char mode = 'A'; mode <= 'z'; mode++)
 	{
 		unsigned char pos = (mode-65) | MASK_CHANNEL;
 
 		if ((modehandlers[pos]) && (modehandlers[pos]->GetPrefix()))
 		{
-			prefixes[modehandlers[pos]->GetPrefixRank()] = std::make_pair(
-				modehandlers[pos]->GetPrefix(), modehandlers[pos]->GetModeChar());
+			prefixes.push_back(modehandlers[pos]);
 		}
 	}
 
-	for(std::map<int,std::pair<char,char> >::reverse_iterator n = prefixes.rbegin(); n != prefixes.rend(); n++)
+	std::sort(prefixes.begin(), prefixes.end(), PrefixModeSorter());
+	for(std::vector<ModeHandler*>::reverse_iterator n = prefixes.rbegin(); n != prefixes.rend(); n++)
```

Comment:
Please make the following changes:
* add space between `for` and `(`
* const_reverse_iterator
* `++n`